tests/rbd/rbd_trash_purge.py

```diff
@@ -0,0 +1,148 @@
+    """
+    Create snap, protect and create clone for the given image
+    Args:
+        rbd: rbd object
+        pool: pool name
+        image_name: image name
+        snap_name: snap name
+        clone_name: clone name
+    """
+    if rbd.snap_create(pool, image_name, snap_name):
+        log.error(f"Snap creation failed for {image_name}")
+        return 1
+    snap_spec = f"{pool}/{image_name}@{snap_name}"
+    if rbd.protect_snapshot(snap_spec):
+        log.error(f"Snap protect failed for {snap_spec}")
+        return 1
+    if rbd.create_clone(snap_spec, pool, clone_name):
+        log.error(f"Clone creation failed for {clone_name}")
+        return 1
+    return 0
+
+
+def rbd_trash_purge(rbd, pool_type, **kw):
+    """
+    Run trash and force remove operation on images
+    Args:
+        rbd: rbd object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: Test data
+    """
+    pool = kw["config"][pool_type]["pool"]
+    num_images = kw["config"].get("no_of_images", 10)
+    num_clones = kw["config"].get("no_of_clones", 4)
+    client = kw["ceph_cluster"].get_nodes(role="client")[0]
+    images_with_clone = list()
+    try:
+        log.info(f"Creating {num_images} number of images in pool {pool}")
+
+        for i in range(0, num_images):
+            image_name = f"image_{i}"
+            if rbd.create_image(pool, image_name, "10G"):
+                log.error(f"Image creation failed for {image_name}")
+                return 1
+            run_fio(
+                image_name=image_name, pool_name=pool, client_node=client, runtime=10
+            )
+
+        log.info(
+            f"Creating snapshot and clone for {num_clones} out of {num_images} number of images"
+        )
+
+        with parallel() as p:
+            for i in range(0, num_clones):
+                image_name = f"image_{i}"
+                images_with_clone.append(image_name)
+                snap_name = f"snap_{i}"
+                clone_name = f"clone_{i}"
+                p.spawn(
+                    create_snap_and_clone, rbd, pool, image_name, snap_name, clone_name
+                )
+
+        log.info("Moving all images created above to trash")
+        for i in range(0, num_images):
+            image_name = f"image_{i}"
+            if rbd.move_image_trash(pool, image_name):
+                log.error(f"Image {image_name} not moved to trash")
+                return 1
+
+        log.info(f"Performing trash purge on pool {pool}")
+        out = rbd.trash_purge(pool)
+
+        if "some expired images could not be removed" in str(out[1]):
+            log.info("rbd trash purge was unable to purge all images in the pool")
+        else:
+            log.error("rbd trash purge did not work as expected")
+            return 1
+
+        for i in range(0, num_images):
+            image = f"image_{i}"
+            if image in images_with_clone and not rbd.trash_exist(pool, image):
```

Comment:
trash_exist executed trash list and we are running same command with same o/p num_images times
IMO, we should pass et of (pool, image) and optimize

Reply:
Right Vasi, makes sense both trash_exist and image_exist should work for either single or multiple images based on the input passed. We can make this change as part of optimizing rbd_utils in a different PR.